--- a/src/core/lib/event_engine/ares_resolver.cc
+++ b/src/core/lib/event_engine/ares_resolver.cc
@@ -371,1 +371,1 @@
-  ares_query(channel_, std::string(host).c_str(), ns_c_in, ns_t_srv,
+  ares_query(channel_, std::string(host).c_str(), C_IN, T_SRV,
@@ -406,1 +406,1 @@
-  ares_search(channel_, std::string(host).c_str(), ns_c_in, ns_t_txt,
+  ares_search(channel_, std::string(host).c_str(), C_IN, T_TXT,
