diff --git a/Makefile b/Makefile
index f308b4c..ec5633d 100644
--- a/Makefile
+++ b/Makefile
@@ -21,24 +21,24 @@ EX_PKG=-not -path "./pkg/*"
 EX_CMD=-not -path "./cmd/*"
 
 # NONROOT_MODS includes ./* dirs (excludes . dir)
-NONROOT_MODS := $(shell find . $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+NONROOT_MODS := $(shell gfind . $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
 
-RECEIVER_MODS_0 := $(shell find ./receiver/[a-k]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-RECEIVER_MODS_1 := $(shell find ./receiver/[l-z]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+RECEIVER_MODS_0 := $(shell gfind ./receiver/[a-k]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+RECEIVER_MODS_1 := $(shell gfind ./receiver/[l-z]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
 RECEIVER_MODS := $(RECEIVER_MODS_0) $(RECEIVER_MODS_1)
-PROCESSOR_MODS := $(shell find ./processor/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-EXPORTER_MODS := $(shell find ./exporter/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-EXTENSION_MODS := $(shell find ./extension/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-CONNECTOR_MODS := $(shell find ./connector/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-INTERNAL_MODS := $(shell find ./internal/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-PKG_MODS := $(shell find ./pkg/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-CMD_MODS := $(shell find ./cmd/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-OTHER_MODS := $(shell find . $(EX_COMPONENTS) $(EX_INTERNAL) $(EX_PKG) $(EX_CMD) $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) ) $(PWD)
+PROCESSOR_MODS := $(shell gfind ./processor/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+EXPORTER_MODS := $(shell gfind ./exporter/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+EXTENSION_MODS := $(shell gfind ./extension/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+CONNECTOR_MODS := $(shell gfind ./connector/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+INTERNAL_MODS := $(shell gfind ./internal/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+PKG_MODS := $(shell gfind ./pkg/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+CMD_MODS := $(shell gfind ./cmd/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+OTHER_MODS := $(shell gfind . $(EX_COMPONENTS) $(EX_INTERNAL) $(EX_PKG) $(EX_CMD) $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) ) $(PWD)
 ALL_MODS := $(RECEIVER_MODS) $(PROCESSOR_MODS) $(EXPORTER_MODS) $(EXTENSION_MODS) $(CONNECTOR_MODS) $(INTERNAL_MODS) $(PKG_MODS) $(CMD_MODS) $(OTHER_MODS)
 
 # find -exec dirname cannot be used to process multiple matching patterns
-FIND_INTEGRATION_TEST_MODS={ find . -type f -name "*integration_test.go" & find . -type f -name "*e2e_test.go" -not -path "./testbed/*"; }
-INTEGRATION_MODS := $(shell $(FIND_INTEGRATION_TEST_MODS) | xargs $(TO_MOD_DIR) | uniq)
+FIND_INTEGRATION_TEST_MODS={ gfind . -type f -name "*integration_test.go" & gfind . -type f -name "*e2e_test.go" -not -path "./testbed/*"; }
+INTEGRATION_MODS := $(shell $(FIND_INTEGRATION_TEST_MODS) | gxargs $(TO_MOD_DIR) | uniq)
 
 ifeq ($(GOOS),windows)
 	EXTENSION := .exe
@@ -384,11 +384,11 @@ crosslink: $(CROSSLINK)
 .PHONY: clean
 clean:
 	@echo "Removing coverage files"
-	find . -type f -name 'coverage.txt' -delete
-	find . -type f -name 'coverage.html' -delete
-	find . -type f -name 'coverage.out' -delete
-	find . -type f -name 'integration-coverage.txt' -delete
-	find . -type f -name 'integration-coverage.html' -delete
+	gfind . -type f -name 'coverage.txt' -delete
+	gfind . -type f -name 'coverage.html' -delete
+	gfind . -type f -name 'coverage.out' -delete
+	gfind . -type f -name 'integration-coverage.txt' -delete
+	gfind . -type f -name 'integration-coverage.html' -delete
 
 .PHONY: genconfigdocs
 genconfigdocs:
diff --git a/Makefile.Common b/Makefile.Common
index fae4d9c..5c68393 100644
--- a/Makefile.Common
+++ b/Makefile.Common
@@ -2,7 +2,7 @@
 # the value `-o pipefail` (or `set -o pipefail`) is added to each shell command that make runs
 # otherwise in the example command pipe, only the exit code of `tee` is recorded instead of `go test` which can cause
 # test to pass in CI when they should not.
-SHELL = /bin/bash
+SHELL = /bin/sh
 .SHELLFLAGS = -o pipefail -c
 
 SHELL_CASE_EXP = case "$$(uname -s)" in CYGWIN*|MINGW*|MSYS*) echo "true";; esac;
